### ISSUE TITLE
Time broken in 2023.05.01

### ISSUE THREAD

**Issue by igorzhitko** [joncampbell123/dosbox-x#4232]
### Describe the bug

I noticed in one program that seconds jump back and forth in time. I wrote a small test program and received such confirmation.
cut from dosbox.log
DOS CON: C:\WORK\TESTS>testtime
Flush STDIN
DOS CON: 23:23:47 (824)
DOS CON: 23:23:46 (879)
DOS CON: 23:23:47 (934)
DOS CON: 23:23:46 (989)
DOS CON: 23:23:47 (1044)
DOS CON: 23:23:46 (1099)
DOS CON: 23:23:48 (1813)
DOS CON: 23:23:47 (1868)
DOS CON: 23:23:48 (1923)
DOS CON: 23:23:47 (1978)
DOS CON: 

### Steps to reproduce the behaviour

Compile and run
#include <stdio.h>
#include <time.h>

int main( int, char ** ) {
  time_t oldtime = time( NULL );
  short steps = 10;
  while( steps ) {
    while( oldtime == time( NULL ));
    oldtime = time( NULL );
    steps --;
    tm * td = localtime( &oldtime );
    printf( "%02d:%02d:%02d (%ld)\n", td->tm_hour, td->tm_min, td->tm_sec, clock());
  }
}

I use OpenWatcom, where clock() returns thousandths of a second

### Expected behavior

Same program run on dosbox-x 2023.03.31
cut from dosbox.log
DOS CON: C:\WORK\TESTS>testtime
Flush STDIN
DOS CON: 23:37:53 (879)
DOS CON: 23:37:54 (1867)
DOS CON: 23:37:55 (2856)
DOS CON: 23:37:56 (3899)
DOS CON: 23:37:57 (4888)
DOS CON: 23:37:58 (5877)
DOS CON: 23:37:59 (6865)
DOS CON: 23:38:00 (7854)
DOS CON: 23:38:01 (8898)
DOS CON: 23:38:02 (9886)
DOS CON: 

### What operating system(s) this bug have occurred on?

Windows 11 Pro x64

### What version(s) of DOSBox-X have this bug?

2023.05.01 x64 Release

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by joncampbell123** [comment]
Well at least I've cleaned up the DOS interface to call the BIOS, and the BIOS to read/write the RTC, so what should actually happen is the RTC should tick along at the emulator speed according to how an RTC would count time. It would make it much more accurate according to the RTC interrupt bits and allow time to appear normally to the guest even if the emulator can't run at 100% speed. It should of course initialize the time according to the host time at emulator startup or perhaps other user initiated times.

**Comment by joncampbell123** [comment]
Pulling out the 486 laptop, I just found something quite interesting.

Apparently DOS reads the RTC once at startup, and then time and date are maintained as IRQ0 timer ticks from that point on.

You can "stop the clock" for DOS by masking IRQ0.

When you do that, there's a very obvious discrepancy between what INT 21h says the time is vs INT 1Ah.

Interesting.

**Comment by joncampbell123** [comment]
For the PC-98 case I'll have to see if it really does read the RTC on INT 21h because IRQ0 on PC-98 is always masked off unless a timer interval is desired by a DOS program or driver.

**Comment by joncampbell123** [comment]
OK. CMOS emulation has been updated to tick at emulation time, BIOS to read/write it properly and init on startup, and DOS to handle date and time as DOS apparently does. If you can compile from source, please try the latest commit.

**Comment by igorzhitko** [comment]
I don't have Visual Studio anymore. I tried build from Actions menu.
I checked both program and I see that the problrm is gone.
cut from dosbox.log
DOS CON: C:\WORK\N-AZS\TESTS>testtime
Flush STDIN
DOS CON: 10:30:47 (824)
DOS CON: 10:30:48 (1813)
DOS CON: 10:30:49 (2857)
DOS CON: 10:30:50 (3845)
DOS CON: 10:30:51 (4834)
DOS CON: 10:30:52 (5823)
DOS CON: 10:30:53 (6866)
DOS CON: 10:30:54 (7855)
DOS CON: 10:30:55 (8843)
DOS CON: 10:30:56 (9832)
DOS CON: 

There was another problem that the buttons worked jerkily, for example, scrolling through a directory in nc. This "effect" is also no longer repeated.
